v2/internal/reconcilers/arm/azure_generic_arm_reconciler.go

```diff
@@ -38,8 +38,9 @@ const (
 type DeleteAction string
 
 const (
-	DeleteActionBeginDelete   = DeleteAction("BeginDelete")
-	DeleteActionMonitorDelete = DeleteAction("MonitorDelete")
+	DeleteActionBeginDelete        = DeleteAction("BeginDelete")
+	DeleteActionMonitorDelete      = DeleteAction("MonitorDelete")
+	DeleteActionNotPossibleInAzure = DeleteAction("NotPossibleInAzure")
```

Comment:
Why add another "possible action" here, rather than just doing the check at the start if `StartDeleteOfResource` (or possibly `deleteResource` helper since then we could write an extension to override that behavior if we wanted?)